packages/react/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,233 @@
+  defaultOpen?: boolean
+
+  /** A split button can be disabled. */
+  disabled?: boolean
+
+  /** Shorthand for the menu. */
+  menu?: ShorthandValue<MenuProps> | ShorthandCollection<MenuItemProps>
+
+  /**
+   * Called after user's click on the main button.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onMainButtonClick?: ComponentEventHandler<ButtonProps>
+
+  /**
+   * Called after user's click on a menu item.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onMenuItemClick?: ComponentEventHandler<MenuItemProps>
+
+  /**
+   * Event for request to change 'open' value.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onOpenChange?: ComponentEventHandler<PopupProps>
+
+  /** Defines whether menu is displayed. */
+  open?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  primary?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  secondary?: boolean
+
+  /** Shorthand for the toggle button. */
+  toggleButton?: ShorthandValue<ButtonProps>
+}
+
+export interface SplitButtonState {
+  isFromKeyboard: boolean
+  open: boolean
+}
+
+class SplitButton extends AutoControlledComponent<WithAsProp<SplitButtonProps>, SplitButtonState> {
+  static create: Function
+
+  static displayName = 'SplitButton'
+
+  static className = 'ui-splitbutton'
+
+  static slotClassNames: SplitButtonSlotClassNames = {
+    toggleButton: `${SplitButton.className}__toggleButton`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    button: customPropTypes.itemShorthand,
+    defaultOpen: PropTypes.bool,
+    menu: PropTypes.oneOfType([
+      customPropTypes.itemShorthand,
+      PropTypes.arrayOf(customPropTypes.itemShorthandWithoutJSX),
+    ]),
+    onMainButtonClick: PropTypes.func,
+    onMenuItemClick: PropTypes.func,
+    onOpenChange: PropTypes.func,
+    open: PropTypes.bool,
+    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
+    toggleButton: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    accessibility: splitButtonBehavior,
+    as: 'span',
+    toggleButton: {},
+  }
+
+  static autoControlledProps = ['open']
+
+  getInitialAutoControlledState(): SplitButtonState {
+    return {
+      isFromKeyboard: false,
+      open: false,
+    }
+  }
+
+  handleMenuButtonOverrides = (predefinedProps: MenuButtonProps) => ({
+    onMenuItemClick: (e: React.SyntheticEvent, menuItemProps: MenuItemProps) => {
+      this.setState({ open: false })
+      _.invoke(this.props, 'onOpenChange', e, { open: false, ...this.props })
```

Comment:
```suggestion
      _.invoke(this.props, 'onOpenChange', e, { ...this.props, open: false })
```

Order should be change otherwise a value from `props` will win